common/types.go

```diff
@@ -150,13 +150,10 @@ func HexToAddress(s string) Address    { return BytesToAddress(FromHex(s)) }
 // IsHexAddress verifies whether a string can represent a valid hex-encoded
 // Ethereum address or not.
 func IsHexAddress(s string) bool {
-	if len(s) == 2+2*AddressLength && IsHex(s) {
-		return true
+	if HasHexPrefix(s) {
+		return len(s) == 2+2*AddressLength && IsHex(s)
```

Comment:
This could be `s = s[2:]` to avoid repeating the check expression.

Reply:
True.